### PR TITLE
[Bug] Correct the transformation calculation of views

### LINKED ISSUE #11931: [Bug] [UWP] View translation is incorrectly calculated
### Description

When the translation offset for a view is calculated, it incorrectly uses the scale to multiply/divide the actual offset.

Side bug, seems WPF does not scale at all.

### Steps to Reproduce

1. Create a new project
2. Replace the contents of the main page with this: https://gist.github.com/mattleibow/641614e900cf35bdf5decc13c15ff215
3. Observe that UWP is not quite like the others

### Expected Behavior

The buttons are moved and then scaled independently.

### Actual Behavior

On UWP, the translation also seems to be scaled.

Seems to be this: https://github.com/xamarin/Xamarin.Forms/blob/c003c1a3230a104ac27ee3811b84ebd7a10a2699/Xamarin.Forms.Platform.UAP/VisualElementTracker.cs#L710-L734

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

I added red blocks to where things are supposed to be.

| Android | iOS | UWP | WPF |
| :-: | :-: | :-: | :-: |
| image | image | image | image |

### Reproduction Link

[ViewScalingTranslatingTests.zip](https://github.com/xamarin/Xamarin.Forms/files/5131156/ViewScalingTranslatingTests.zip)

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change ###

When the translation offset for a view is calculated, it incorrectly uses the scale to multiply/divide the actual offset. Side bug, seems WPF does not scale at all.

This PR will correct the math, I hope.

### Issues Resolved ### 

- Fixes #11931

### API Changes ###

 
 None

### Platforms Affected ### 

- iOS: the rotation now happens before the scale, and thus the scale matrix is no longer being rotated
- UWP: the scale no longer affectes the translate
- WPF: added support for scaling along a single axis, and rotate before the scale

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

| Android| iOS | UWP | WPF |
| :-: | :-: | :-: | :-: |
| | | | |
| **BEFORE** | **BEFORE** | **BEFORE** | **BEFORE** |
| android | ios-before | uwp-before | wpf-before |
| **AFTER** | **AFTER** | **AFTER** | **AFTER** |
|  | ios-after | uwp-after | wpf-after |

Not applicable

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by mattleibow** [comment]
I may need help to see if this is actually UI testable... The scaling doesn't affect the bounds properties, so it is going to be a bit harder to check.

**Comment by mattleibow** [comment]
BONUS: Fix the WPF rendering and also the iOS transform order... See screenshots :)

**Comment by mattleibow** [comment]
I did not test Tizen and macOS yet.

**Comment by rookiejava** [comment]
@mattleibow Tizen's result is here. I think it's OK.
translation

**Comment by mattleibow** [comment]
@rookiejava I think the rotation is distorting the text - see the before/after for iOS.

I think the order of scale and rotation are swapped...

**Comment by rookiejava** [comment]
> I think the rotation is distorting the text - see the before/after for iOS.
> I think the order of scale and rotation are swapped...

@mattleibow gotcha! 

As you said, we need tp reverse the order of the two lines below. ( applyScale => applyRotation)

https://github.com/xamarin/Xamarin.Forms/blob/c5e8e31fa559c7632d2ff5209f199771e6d32ed1/Xamarin.Forms.Platform.Tizen/Renderers/VisualElementRenderer.cs#L1090-L1091

The result is as below.

**BEFORE** | **AFTER**
--- | --- 
 |  

Thanks! 😉

**Comment by mattleibow** [comment]
Awesome! Could you crazy a quick or for that onto my branch, and then I'll merge when I wake up?

**Comment by rookiejava** [comment]
> Awesome! Could you create a quick pr for that onto my branch, and then I'll merge when I wake up?

Done. Thanks!

**Comment by mattleibow** [comment]
Merged! Thanks.

I think this PR is ready!!!

**Review by PureWeen — CHANGES_REQUESTED** [review]
@mattleibow is writing Unit Tests

File: Xamarin.Forms.Platform.UAP.UnitTests/TransformationTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+					element.Scale = 2.0;
+					element.ScaleX = 2.0;
+					yield return CreateTestCase(element);
+				}
+			}
+		}
+
+		[Test, Category("Transformation"), TestCaseSource(nameof(TransformationCases))]
+		[Description("View transformation should match renderer transformation")]
+		public async Task TransformationConsistent(View view)
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Platform.UAP.UnitTests/TransformationTests.cs:31]: :heart:

**Review by jsuarezruiz — APPROVED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]
Looks like it still needs a little bit more rounding love @mattleibow 

image

**Comment by mattleibow** [comment]
I am rounding to 3 decimal places... Maybe the library has a bug or something? Maybe it is the nfloat or something.

I'll check.

**Comment by mattleibow** [comment]
I see lots more green, but still a bit of red...

**Comment by rmarinho** [comment]
The iOS failure is not related.

**Review by PureWeen — APPROVED** [review]

**Comment by mattleibow** [comment]
Another place were this actually shows up as what I think i an improvement is the "pinch gallery". Before, when you pinch, it would scale from the left edge. Afterwards, it scales from the center as expected.

**Pull request merged**